annubes/tasks/task.py

```diff
@@ -1,192 +1,205 @@
 import colorsys
-import random
+import warnings
 from collections import OrderedDict
+from dataclasses import dataclass, field
 
 import numpy as np
 import plotly.graph_objects as go
 from numpy.typing import NDArray
 from plotly.subplots import make_subplots
 
 
+@dataclass
 class Task:
-    """General class for defining a task."""
+    """General class for defining a task.
+
+    Args:
+        name: Name of the task.
+        session: Dictionary representing the ratio (values) of the different trials (keys) within the task. Trials with
+            a single modality must be represented by single characters, while trials with multiple modalities are
+            represented by the character combination of those trials. Note that the order of the dictionary is
+            maintained such that `{"v": 0.5, "a": 0.5}` == `{"a": 0.5, "v": 0.5}` is False. Note also that values are
+            read relative to each other, such that e.g. `{"v": 0.25, "a": 0.75}` == `{"v": 1, "a": 3}` is True.
+            Defaults to {"v": 0.5, "a": 0.5}.
+        stim_intensities: List of possible intensities of each stimulus. Note that this
+            attribute will be sorted smallest to largest. Defaults to [0.8, 0.9, 1].
+        catch_prob: probability of catch trials in the session, between 0 and 1.
+            Defaults to 0.5.
+        catch_intensity: Intensity value during a catch trial.
```

Comment:
Oh yes now I remember where this comes from. Okay so it was about the input signals, not the output ones. Jorge indeed said that it may be useful to be able to set the input signals to a certain value (different from 0) during catch trials. Then I think that `catch_intensity` can work, but it should be used for filling in the x signal instead of the 0s (if different from 0). I don't think we should use it for the output signals.